File: mobile/lib/model/common/order_status.dart
enum OrderStatus {
  pending,
  confirmed,
  packing,
  shipped,
  delivered,
  completed,
  cancelled;

  static OrderStatus fromApi(String? value) {
    return switch (value?.toUpperCase()) {
      'PENDING' => OrderStatus.pending,
      'PAID' => OrderStatus.confirmed,
      'CONFIRMED' => OrderStatus.confirmed,
      'PACKING' => OrderStatus.packing,
      'SHIPPING' => OrderStatus.shipped,
      'SHIPPED' => OrderStatus.shipped,
      'DELIVERED' => OrderStatus.delivered,
      'COMPLETED' => OrderStatus.completed,
      'CANCELLED' => OrderStatus.cancelled,
      _ => OrderStatus.pending,
    };
  }

  String get apiValue {
    return switch (this) {
      OrderStatus.pending => 'PENDING',
      OrderStatus.confirmed => 'CONFIRMED',
      OrderStatus.packing => 'PACKING',
      OrderStatus.shipped => 'SHIPPED',
      OrderStatus.delivered => 'DELIVERED',
      OrderStatus.completed => 'COMPLETED',
      OrderStatus.cancelled => 'CANCELLED',
    };
  }
}

File: backend/order-fulfillment-service/src/main/java/org/example/model/enums/OrderStatus.java
package org.example.model.enums;

public enum OrderStatus {
    PENDING,    // Chờ xử lý/Chờ thanh toán
    CONFIRMED,  // Đã xác nhận
    PACKING,    // Đang đóng gói
    SHIPPED,    // Đã bàn giao cho giao hàng
    COMPLETED,  // Đơn hàng hoàn tất
    CANCELLED,  // Đã hủy

    // Legacy web/payment statuses kept for backward compatibility.
    PAID,       // Đã thanh toán (Chờ giao)
    SHIPPING,    // Đang giao hàng
    DELIVERED  // Giao thành công
}
